pkgs/tools/security/step-kms-plugin/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, pkg-config
+, pcsclite
+, softhsm
+, writeScriptBin }:
+
+let
+  getScriptName = "get_softhsm_so_path";
+  getSoftHSMsoPath = writeScriptBin getScriptName ''
```

Comment:
Why do we need this?

Reply:
When using the tool, you have to give it the path to the .so files: https://github.com/smallstep/step-kms-plugin#general-usage

I have an updated version that works with `opensc` too. Going to try and get all of the various HSMs added.

Reply:
`step kms create --kty EC --crv P384 'pkcs11:module-path=/usr/local/lib/softhsm/libsofthsm2.so;token=smallstep;id=1000;object=mykey?pin-value=password'` <-- example so you don't need to load the page :P

Reply:
We can't just hardcode that?

Reply:
Hard code the `module-path`? No, because it's a constructed (by the end user) URI that will change depending on the HSM used.. or various other user-specified params.